modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/ConnectionManager.java

```diff
@@ -379,6 +379,7 @@ private void closeConnections() {
 
         threadConns.clear();
         detachedConns.clear();
+        threadConn.remove();
```

Comment:
Hello!

Can we add a test for that please? I guess it should be pretty straightforward.